src/components/Layout/Layout.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from "react";
+import Counter from "../Counter/Counter";
+import { Outlet, useSearchParams } from "react-router-dom";
+import MovieListPage from "../MovieListPage/MovieListPage";
+import sortOptions from "../../helpers/sortOptions";
+import { getMovies } from "../../services";
+import genres from "../../helpers/genres";
+import "./Layout.css";
+
+const Layout = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [orderedMovies, setOrderedMovies] = useState([]);
+  const query = searchParams.get("query") || "";
+  const genre = searchParams.get("genre") || genres[0];
+  const sortBy = searchParams.get("sortBy") || sortOptions[0];
+
+  const handleSearch = (value) => {
+    setSearchParams({ ...Object.fromEntries(searchParams), query: value });
+  };
+
+  const handleGenre = (value) => {
+    setSearchParams({ ...Object.fromEntries(searchParams), genre: value });
+  };
+
+  const handleSortChange = (value) => {
+    setSearchParams({ ...Object.fromEntries(searchParams), sortBy: value });
+  };
+
+  useEffect(() => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+    getMovies(sortBy, query, genre === "All" ? "" : genre, signal)
+      .then((response) => setOrderedMovies(response.data))
+      .catch((error) => {
+        if (error.name !== "AbortError") {
+          console.error("Fetch error:", error.message);
```

Comment:
After running all test, this error popsout:

> Fetch error: invalid json response body at  reason: Unexpected end of JSON input

![image](https://github.com/user-attachments/assets/075f3291-4631-4257-862b-3fc5971cdc68)

Reply:
it is expected behavior since there was a task in one of the modules to test concurrency tracking, multiple requests at the same time. The log is for information during tests.
Now I will comment or remove the console error log.